repos/os/include/i2c_session/i2c_session.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * \brief  I2C session client implementation
+ * \author Jean-Adrien Domage <jean-adrien.domage@gapfruit.com>
+ * \date   2021-02-25
+ */
+
+/*                                                                              
+ * Copyright (C) 2013-2021 Genode Labs GmbH                                     
+ * Copyright (C) 2021 gapfruit AG
+ *
+ * This file is part of the Genode OS framework, which is distributed
+ * under the terms of the GNU Affero General Public License version 3.
+ */
+
+#ifndef _INCLUDE__I2C_SESSION__I2C_SESSION_H_
+#define _INCLUDE__I2C_SESSION__I2C_SESSION_H_
+
+#include <session/session.h>
+#include <base/stdint.h>
+
+namespace I2c {
+
+	class Session;
+
+}
+
+class I2c::Session : public Genode::Session
+{
+public:
+
+	/**
+	 * \noapi
+	 */
+	static const char *service_name() { return "I2c"; }
+
+	enum { CAP_QUOTA = 2 };
+
+	/****************
+	 ** Exception **
+	 ****************/
+
+	class Bus_error   : public Genode::Exception {};
+	class Bus_timeout : public Bus_error {};
+
+	class Bus_arbitration_lost   : public Bus_error {};
+	class Slave_error_no_ack     : public Bus_error {};
+
+	/***********************
+	 ** Session interface **
+	 ***********************/
+
+	virtual void            write_byte(Genode::uint8_t) = 0;
+	virtual Genode::uint8_t read_byte() = 0;
+
+	virtual void             write_word(Genode::uint16_t) = 0;
+	virtual Genode::uint16_t read_word() = 0;
+
+	GENODE_RPC_THROW(Rpc_write_byte, void, write_byte,
+	                 GENODE_TYPE_LIST(Bus_error,
+	                                  Bus_timeout,
+	                                  Bus_arbitration_lost,
+	                                  Slave_error_no_ack),
+	                 Genode::uint8_t);
+	GENODE_RPC_THROW(Rpc_read_byte, Genode::uint8_t, read_byte,
+	                 GENODE_TYPE_LIST(Bus_error,
+	                                  Bus_timeout,
+	                                  Bus_arbitration_lost,
+	                                  Slave_error_no_ack));
+
+	GENODE_RPC_THROW(Rpc_write_word, void, write_word,
+	                 GENODE_TYPE_LIST(Bus_error,
+	                                  Bus_timeout,
+	                                  Bus_arbitration_lost,
+	                                  Slave_error_no_ack),
+	                 Genode::uint16_t);
+	GENODE_RPC_THROW(Rpc_read_word, Genode::uint16_t, read_word,
+	                 GENODE_TYPE_LIST(Bus_error,
+	                                  Bus_timeout,
+	                                  Bus_arbitration_lost,
+	                                  Slave_error_no_ack));
+
+	GENODE_RPC_INTERFACE(Rpc_write_byte, Rpc_read_byte,
+	                     Rpc_write_word, Rpc_read_word);
+};
+
```

Comment:
Please, use `/* */` in all non-temporary comments. This is part of Genode's coding style.